build/Microsoft.DotNet.Cli.Publish.targets

```diff
@@ -0,0 +1,107 @@
+  </Target>
+
+  <Target Name="SetupAzureBlobInformation">
+    <PropertyGroup>
+      <Product>Sdk</Product>
+      <ContainerName>$(STORAGE_CONTAINER)</ContainerName>
+      <CloudDropAccessToken>$(STORAGE_KEY)</CloudDropAccessToken>
+      <CloudDropAccountName>$(STORAGE_ACCOUNT)</CloudDropAccountName>
+      <DotnetBlobRootUrl>https://$(CloudDropAccountName).blob.core.windows.net/$(ContainerName)</DotnetBlobRootUrl>
+    </PropertyGroup>    
+  </Target>
+
+  <Target Name="GatherItemsForPattern">
+    <ItemGroup>
+      <ForPublishing Include="@(GeneratedInstallers)" />
+      <ForPublishing Include="%(GenerateArchivesInputsOutputs.Outputs)" />
+    </ItemGroup>
+
+    <ItemGroup>
+      <ForPublishing>
+        <RelativeBlobPath>$(Product)/$(NugetVersion)/$([System.String]::Copy('%(Filename)%(Extension)').Replace('\' ,'/'))</RelativeBlobPath> 
+      </ForPublishing>
+    </ItemGroup>
+  </Target>  
+
+  <Target Name="PublishDebFilesToDebianRepo" Condition=" '$(OSName)' == 'ubuntu' ">
+    <Error Condition="'$(REPO_ID)' == ''" Text="REPO_ID must be set as an environment variable for debian publishing." />
+
+    <ItemGroup>
+      <SdkInstallerFileItemGroup Include="$(SdkInstallerFile)" />
+    </ItemGroup>
+    <PropertyGroup>
+      <SdkDebianPackageName>dotnet-dev-$(NugetVersion)</SdkDebianPackageName>
+      <SdkDebianUploadUrl>$(DotnetBlobRootUrl)/$(Product)/$(NugetVersion)/%(SdkInstallerFileItemGroup.Filename)%(SdkInstallerFileItemGroup.Extension)</SdkDebianUploadUrl>
+      <DebianUploadJsonFile>$(PackageDirectory)/package_upload.json</DebianUploadJsonFile>
+      <DebianRevisionNumber>1</DebianRevisionNumber>
+
+      <DebianUploadJsonContent>
+        {
+          "name":"$(SdkDebianPackageName)",
+          "version":"$(NugetVersion)-$(DebianRevisionNumber)",
+          "repositoryId":"$(REPO_ID)",
+          "sourceUrl": "$(SdkDebianUploadUrl)"
+        }
+      </DebianUploadJsonContent>
+    </PropertyGroup>
+
+    <Delete Files="$(DebianUploadJsonFile)" />
+    <WriteLinesToFile File="$(DebianUploadJsonFile)" Lines="$(DebianUploadJsonContent)" />
+
+    <Exec Command="$(RepoRoot)/scripts/publish/repoapi_client.sh -addpkg $(DebianUploadJsonFile)" />
+  </Target>
+
+  <Target Name="PublishCliVersionBadge">
+    <ItemGroup>
+      <CliVersionBadgeToUpload Include="$(VersionBadge)" />
+    </ItemGroup>
+
+    <ItemGroup>
+      <CliVersionBadgeToUpload>
+        <RelativeBlobPath>$(Product)/$(NugetVersion)/$([System.String]::Copy('%(Filename)%(Extension)').Replace('\' ,'/'))</RelativeBlobPath>
+      </CliVersionBadgeToUpload>
+    </ItemGroup>
+
+    <UploadToAzure
+      AccountKey="$(CloudDropAccessToken)"
+      AccountName="$(CloudDropAccountName)"
+      ContainerName="$(ContainerName)"
+      Items="@(CliVersionBadgeToUpload)" />
+
+    <SetBlobPropertiesBasedOnFileTypeTask
+      AccountKey="$(CloudDropAccessToken)"
+      AccountName="$(CloudDropAccountName)"
+      Items="@(CliVersionBadgeToUpload)" />
```

Comment:
Are we saying that blob metadata gets set post-upload? Making this non-atomic creates room for a race where someone refreshes our readme.md, gets the badly-metadata'd badges, and clogs the image cache for a few weeks. I think we need to unify upload with metadata setting. Can we just give UploadToAsure a dictionary of key/values to set?